openedx_external_enrollments/external_enrollments/salesforce_external_enrollment.py

```diff
@@ -162,26 +179,24 @@ def _get_program_of_interest_data(self, data, order_lines):
             openedx_user, _ = get_user(email=email)
             request_time = datetime.datetime.utcnow()
             if program:
+                bundle_id = program.get("uuid")
                 related_program = ProgramSalesforceEnrollment.objects.get(  # pylint: disable=no-member
-                    bundle_id=program.get("uuid"),
-                )
-                program_of_interest = related_program.meta
-                program_of_interest["Drupal_ID"] = "enrollment+program+{}+{}".format(
-                    openedx_user.username,
-                    request_time.strftime("%Y/%m/%d-%H:%M:%S"),
+                    bundle_id=bundle_id,
                 )
+                if related_program.meta:
+                    program_of_interest = related_program.meta
+                elif self._is_special_program_type(program.get("type")):
```

Comment:
This way you won't need the method `_is_special_program_type` and also to reduce unnecessary code lines, what do you think?
```suggestion
                elif program.get("type") == CUSTOM_BUNDLE_TYPE:
```